Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Document/ShaderManagementConsoleDocument.cpp

```diff
@@ -215,14 +205,59 @@ namespace ShaderManagementConsole
 
     bool ShaderManagementConsoleDocument::IsModified() const
     {
-        return false;
+        return CanUndo();
```

Comment:
That's exactly what is happening here. In this case, because every change is now an undoable operation, if there is anything in the undo history you can consider the document “modified”. That's generally the case in other tools and editors. For example, if you make a change in a text editor but then “undo it” by hand editing text back to its original value, notepad or Visual Studio or whatever tool will still consider it modified unless you undo all of the changes through they're undo system. Generally, you would not have the unmodified data to do a comparison or want to do that kind of processing just to check if a document was modified. Material editor is the outlier because it does have that data. The more "correct" thing to do would be to have material editor check the undo history exactly like this instead of checking for modified values.

Reply:
If you save the file, is it still "modified". According to this code yes it will still show as modified (unless you are clearing the undo stack). But if I save the file I would not expect the application to indicate that the document is modified.